File: tests/integration/construct_component_unknown/python/component.py
# Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

from typing import Optional

import codeinfra

class Component(codeinfra.ComponentResource):
    def __init__(self, name: str, args: codeinfra.Inputs, opts: Optional[codeinfra.ResourceOptions] = None):
        super().__init__("testcomponent:index:Component", name, args, opts, True)

File: tests/integration/construct_component_error_apply/nodejs/component.ts
// Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

interface ComponentArgs {
    foo: codeinfra.Input<string>;
}

export class Component extends codeinfra.ComponentResource {
    public readonly foo!: codeinfra.Output<string>;

    constructor(name: string, args: ComponentArgs, opts?: codeinfra.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["foo"] = args.foo;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}

